src/scaffoldmaker/meshtypes/meshtype_3d_stellate1.py

```diff
@@ -0,0 +1,543 @@
+                                scalefactors = [-1.0]
+                                remapEftNodeValueLabel(eft1, [4, 8], Node.VALUE_LABEL_D_DS1,
+                                                       [(Node.VALUE_LABEL_D_DS2, [1])])
+                            ln_map = [1, 2, 3, 2, 4, 5, 6, 5]
+                            remapEftLocalNodes(eft1, 6, ln_map)
+
+                        if eft1 is not eft:
+                            elementtemplateX.defineField(coordinates, -1, eft1)
+                            elementtemplate1 = elementtemplateX
+
+                        element = mesh.createElement(elementIdentifier, elementtemplate1)
+                        result = element.setNodesByIdentifier(eft1, nodeIdentifiers)
+                        result3 = element.setScaleFactors(eft1, scalefactors) if scalefactors else None
+                        elementIdentifier += 1
+        fm.endChange()
+        return []
+
+    @classmethod
+    def refineMesh(cls, meshrefinement, options):
+        """
+        Refine source mesh into separate region, with change of basis.
+        :param meshrefinement: MeshRefinement, which knows source and target region.
+        :param options: Dict containing options. See getDefaultOptions().
+        """
+        assert isinstance(meshrefinement, MeshRefinement)
+        refineElementsCount1 = options['Refine number of elements along arm']
+        refineElementsCount2 = options['Refine number of elements across arm']
+        refineElementsCount3 = options['Refine number of elements through thickness']
+        meshrefinement.refineAllElementsCubeStandard3d(refineElementsCount1, refineElementsCount2, refineElementsCount3)
+
+
+def createArm(thAr, elementLengths, armAngle, armAngleConst, elementsCount, dipMultiplier, numArm, na):
+    """
+    Create single arm unit.
+    Base length of element is 1.
+    Direction: anticlockwise.
+    Minimum arm length is 2 elements.
+    :param thAr: angle arising from base node (rad)
+    :param elementLengths: [Element length along arm, half Element width across arm, Element thickness]
+    :param armAngle: angle from x axis of the arm about origin (rad)
+    :param armAngleConst: minimum angle from x axis of the first arm about origin (rad)
+    :param elementsCount: element counts in x,y,z directions
+    :param dipMultiplier: factor that wheel nodes protrude by, relative to unit length
+    :param numArm: number of arms in body
+    :param na: nth arm currently being created
+    :return: x: positions of nodes in arm
+    :return: nodeIdentifier: number of last node in arm +1
+    :return: xnodes_ds1: ds1 derivatives of nodes associated with x
+    :return: xnodes_ds2: ds2 derivatives of nodes associated with y
+    """
+
+    elementsCount1, elementsCount2, elementsCount3 = elementsCount
+    [elen, ewid, zheight] = elementLengths
+
+    xnodes_ds1 = []
+    xnodes_ds2 = []
+    dx_ds1 = [elen, 0.0, 0.0]
+    dx_ds2 = [0.0, ewid, 0.0]
+    wheelDvMult = [0.5, 0.75]
```

Comment:
These factors should be calculated from smoothing around the gap between arms. Can fix it up after the main loop and use smoothCubicHermiteDerivativesLine to fix d1 at the dips.

Reply:
How do I go about replacing this code?